connectors/byoc.py

```diff
@@ -148,69 +146,192 @@ def build_docs_query(self, native_service_types=None, connectors_ids=None):
         return query
 
     def _create_object(self, doc_source):
-        return BYOConnector(
+        return Connector(
             self,
             doc_source["_id"],
             doc_source["_source"],
             bulk_options=self.bulk_options,
             language_code=self.language_code,
-            analysis_icu=self.analysis_icu,
         )
 
 
-class SyncJob:
-    def __init__(self, connector_id, elastic_client):
-        self.connector_id = connector_id
-        self.client = elastic_client
-        self.created_at = datetime.now(timezone.utc)
-        self.completed_at = None
-        self.job_id = None
-        self.status = None
+class SyncJobIndex(ESIndex):
+    READ_ONLY_FIELDS = ("connector", "trigger_method", "created_at")
 
-    @property
-    def duration(self):
-        if self.completed_at is None:
-            return -1
-        msec = (self.completed_at - self.created_at).microseconds
-        return round(msec / 9, 2)
+    def __init__(self, elastic_config):
+        super().__init__(index_name=JOBS_INDEX, elastic_config=elastic_config)
 
-    async def start(self, trigger_method=JobTriggerMethod.SCHEDULED):
-        self.status = JobStatus.IN_PROGRESS
+    def build_docs_query(self, native_service_types=None, connector_ids=None):
+        if native_service_types is None:
+            native_service_types = []
+        if connector_ids is None:
+            connector_ids = []
+
+        if len(native_service_types) == 0 and len(connector_ids) == 0:
+            logger.info(
+                "Attempting to fetch jobs without specifying supported types or connector ids is not possible"
+            )
+            return
+        subqueries = []
+
+        subqueries.append(
+            {
+                "bool": {
+                    "must": [
+                        {
+                            "terms": {
+                                "status": [
+                                    e2str(JobStatus.PENDING),
+                                    e2str(JobStatus.SUSPENDED),
+                                ]
+                            }
+                        }
+                    ]
+                }
+            }
+        )
+
+        if len(native_service_types) > 0:
+            subqueries.append(
+                {
+                    "bool": {
+                        "filter": [
+                            {"terms": {"service_type": native_service_types}},
+                        ]
+                    }
+                }
+            )
+
+        if len(connector_ids) > 0:
+            subqueries.append(
+                {
+                    "bool": {
+                        "filter": [
+                            {"terms": {"connector.id": connector_ids}},
+                        ]
+                    }
+                }
+            )
+
+        query = {"bool": {"should": subqueries}}
```

Comment:
This won't work as expected. For a `should` clause, it will match if any of the clause matches.